### ISSUE TITLE
[CS2] Lexer/Nodes: Expansion that becomes rest parameter causes runtime error

### ISSUE THREAD

**Issue by Inve1951** [jashkenas/coffeescript#4631]
copy pasting from actual code
```coffee
header = header[(6 + header.lastIndexOf "/\/ME"+"TA") ... header.lastIndexOf "*/\/"]  # BD EOL bug workaround
```
used to compile to: (cs2-beta3)
```JS
header = header.slice(6 + header.lastIndexOf("/\/ME" + "TA"), header.lastIndexOf("*/\/"));
```
output with beta4:
```js
header = header[(6 + header.lastIndexOf("/\/ME" + "TA"))(...header.lastIndexOf("*/\/"))];
```

now throws at runtime: `TypeError: (var)[Symbol.iterator] is not a function(…)`
also the supposed comment is missing in the js

**Comment by GeoffreyBooth** [comment]
@zdenko do you mind taking a look at this?

**Comment by zdenko** [comment]
@GeoffreyBooth sure.

**Comment by GeoffreyBooth** [comment]
Fixed by #4634.
